hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -730,85 +800,76 @@ onOpenNetworkAckSn
 --
 -- __Transition__: 'OpenState' → 'OpenState'
 onOpenClientClose ::
-  -- | Current chain state
-  ChainStateType tx ->
-  ConfirmedSnapshot tx ->
+  OpenState tx ->
   Outcome tx
-onOpenClientClose chainState confirmedSnapshot =
+onOpenClientClose st =
   OnlyEffects [OnChainEffect{chainState, postChainTx = CloseTx confirmedSnapshot}]
+ where
+  CoordinatedHeadState{confirmedSnapshot} = coordinatedHeadState
+
+  OpenState{chainState, coordinatedHeadState} = st
 
 -- | Observe a close transaction. If the closed snapshot number is smaller than
 -- our last confirmed, we post a contest transaction. Also, we do schedule a
 -- notification for clients to fanout at the deadline.
 --
 -- __Transition__: 'OpenState' → 'ClosedState'
 onOpenChainCloseTx ::
-  HeadParameters ->
-  -- | Current head state; recorded as previous recoverable state
-  HeadState tx ->
-  -- | New chain state
+  OpenState tx ->
+  -- | New chain state.
   ChainStateType tx ->
-  -- | The offchain coordinated head state
-  CoordinatedHeadState tx ->
+  -- | Closed snapshot number.
   SnapshotNumber ->
+  -- | Contestation deadline.
   UTCTime ->
-  HeadId ->
   Outcome tx
-onOpenChainCloseTx
-  parameters
-  headState
-  newChainState
-  coordinatedHeadState
-  closedSnapshotNumber
-  contestationDeadline
-  headId =
-    -- TODO(2): In principle here, we want to:
-    --
-    --   a) Warn the user about a close tx outside of an open state
-    --   b) Move to close state, using information from the close tx
-    NewState closedState $
-      ClientEffect headIsClosed :
-        [ OnChainEffect
-          { -- XXX: Field access on sum-type.
-            chainState = getField @"chainState" headState
-          , postChainTx = ContestTx{confirmedSnapshot}
-          }
-        | onChainEffectCondition
-        ]
-   where
-    CoordinatedHeadState{confirmedSnapshot} =
-      coordinatedHeadState
-    closedState =
+onOpenChainCloseTx openState newChainState closedSnapshotNumber contestationDeadline =
+  NewState closedState $
+    notifyClient :
+      [ OnChainEffect
+        { -- REVIEW: Was using "old" chainState before
```

Comment:
Is using the `newChainState` here correct as we were using the previous one before?

Reply:
Not sure yet, that's why I left this comment 🙂 It might have been an oversight and is more evident now